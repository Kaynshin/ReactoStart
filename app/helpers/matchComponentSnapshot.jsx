/* global expect*/
import renderer from 'react-test-renderer';
import React from 'react';

/**
 * @description HO function that accept Component on first call and props object as second call
 * @param Component
 */
const matchComponentSnapshot = Component => props => {
  const rendered = renderer.create(
    <Component {...props} />
  ).toJSON();
  expect(rendered).toMatchSnapshot();
};

export default matchComponentSnapshot;
