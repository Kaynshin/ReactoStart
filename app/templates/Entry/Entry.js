// --- npm ---
import React from 'react';

// --- scss ---
import './Entry.scss';

const Entry = () => (
  <div className="Entry__container">
    <h1>Hello World!</h1>
  </div>
);

export default Entry;
