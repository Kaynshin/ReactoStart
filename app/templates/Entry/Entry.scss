// --- inclues ---
@import '../../assets/styles/inc.scss';

.Entry__container {
  color: $color_blue;
}
