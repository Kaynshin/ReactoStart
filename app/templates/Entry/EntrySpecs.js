/* global jest */
/* global describe */
/* global it */

// --- components ---
import Entry from './Entry';

// --- helpers ---
import matchComponentSnapshot from '../../helpers/matchComponentSnapshot';

const matchFooterSnapshotComponent = matchComponentSnapshot(Entry);

describe('Entry container', () => {
  it('should render Entry', () => {
    matchFooterSnapshotComponent();
  });
});
