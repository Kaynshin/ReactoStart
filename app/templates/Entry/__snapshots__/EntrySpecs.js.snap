exports[`Entry container should render Entry 1`] = `
<div
  className="Entry__container">
  <h1>
    Hello World!
  </h1>
</div>
`;
